Assign window.ethereum to givenProvider when available (#3378)
ethereum/web3.js: docs/getting-started.rst

- pure js: link the ``dist/web3.min.js``

After that you need to create a web3 instance and set a provider.

> Ethereum supported Browsers like Mist or MetaMask will have a ``ethereumProvider`` or ``ethereum``

Reviewer: We can probably remove the reference to `Mist` here since it is deprecated. Can add e.g. "Trust Wallet" which injects a provider.

Author: Corrected in a1750bd

Reviewer: thx!

> Ethereum supported Browsers like Mist or MetaMask will have a ``ethereumProvider`` or ``ethereum``

Author: ```suggestion
Ethereum supported Browsers like TrustWallet or MetaMask will have an ``ethereumProvider`` or ``ethereum``
```

> Ethereum supported Browsers like TrustWallet or MetaMask will have an ``ethereumProvider`` or ``ethereum``

Reviewer: Hm, could be reworded like:
> Most Ethereum-supported browsers like MetaMask and TrustWallet have an [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compliant provider available at `window.ethereum`. For web3.js, check `Web3.givenProvider`.

edit: added in https://github.com/ethereum/web3.js/pull/3378/commits/0223059d04e4a777cf864029c2c108e3f7ff56e9
